project/SparkBuild.scala

```diff
@@ -119,7 +119,7 @@ object SparkBuild extends PomBuild {
   lazy val publishLocalBoth = TaskKey[Unit]("publish-local", "publish local for m2 and ivy")
 
   lazy val sharedSettings = graphSettings ++ genjavadocSettings ++ Seq (
-    javaHome   := Properties.envOrNone("JAVA_HOME").map(file),
+    javaHome   := sys.props.get("java.home").map(file),
```

Comment:
I'm testing out this patch on one of the Jenkins boxes and it looks like this ended up breaking compilation because `java.home` points to a JRE, which will cause us to fail with an error when SBT tries to shell out to `javac`:

```
java.io.IOException: Cannot run program "/usr/java/jdk1.7.0_71/jre/bin/javac": error=2, No such file or directory
	at java.lang.ProcessBuilder.start(ProcessBuilder.java:1047)
	at sbt.SimpleProcessBuilder.run(ProcessImpl.scala:349)
	at sbt.AbstractProcessBuilder.run(ProcessImpl.scala:128)
	at sbt.AbstractProcessBuilder$$anonfun$runBuffered$1.apply(ProcessImpl.scala:159)
	at sbt.AbstractProcessBuilder$$anonfun$runBuffered$1.apply(ProcessImpl.scala:159)
	at sbt.compiler.JavacLogger.buffer(AggressiveCompile.scala:235)
	at sbt.AbstractProcessBuilder.runBuffered(ProcessImpl.scala:159)
	at sbt.AbstractProcessBuilder.$bang(ProcessImpl.scala:156)
[...]
```